src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDmOptional = userService.findByEmail(email);
+        if (userFromDmOptional.isEmpty()) {
+            throw new AuthenticationException("Can`t login user with email " + email,
+                    new RuntimeException());
+        }
+        User user = userFromDmOptional.get();
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (hashedPassword.equals(user.getPassword())) {
+            return user;
+        }
+        throw new AuthenticationException("Can`t login user with email " + email,
+                new RuntimeException());
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        User user = new User(email, password);
+        userService.add(user);
+        return user;
+    }
```

Comment:
where do you throw registration exception? 
![image](https://github.com/mate-academy/hibernate-user-service-hw/assets/49767049/7131d88c-5816-46a5-adf2-abc1e7a914f4)